lib/engine/game/g_1880/step/buy_train.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/buy_train'
+
+module Engine
+  module Game
+    module G1880
+      module Step
+        class BuyTrain < Engine::Step::BuyTrain
+          def process_buy_train(action)
+            train = action.train
+            @game.train_marker = action.entity if train.owner == @game.depot
+            super
+            @round.bought_trains = true
+          end
+
+          def round_state
+            { bought_trains: false }
+          end
+
+          def pass!
+            super
+            @round.bought_trains = false
```

Comment:
There is a setup method that gets called after each company is done operating. This is a good place to reset state. https://github.com/tobymao/18xx/blob/master/lib/engine/round/operating.rb#L69